crates/sui-core/src/authority/authority_store.rs

```diff
@@ -1895,6 +2005,34 @@ impl ChildObjectResolver for AuthorityStore {
         }
         Ok(Some(child_object))
     }
+
+    fn receive_object_at_version(
+        &self,
+        owner: &ObjectID,
+        receiving_object_id: &ObjectID,
+        receive_object_at_version: SequenceNumber,
+        epoch_id: EpochId,
+    ) -> SuiResult<Option<Object>> {
+        let Some(recv_object) = self.get_object_by_key(receiving_object_id, receive_object_at_version)? else {
+            return Ok(None)
+        };
+
+        // Invalid access -- treat as the object does not exist.
+        if recv_object.owner != Owner::AddressOwner((*owner).into()) {
+            return Ok(None);
+        }
+
+        // If we've already received the object at the version, then we say that it doesn't exist.
+        if self.have_received_object_at_version(
+            receiving_object_id,
+            receive_object_at_version,
+            epoch_id,
+        )? {
+            return Ok(None);
```

Comment:
can you combine these two ifs into a single if statement, and add a comment explaining that we rely on these two cases being indistinguishable for correctness?